ironfish-rust/src/transaction/mod.rs

```diff
@@ -91,30 +84,18 @@ impl ProposedTransaction {
     }
 
     /// Spend the note owned by spender_key at the given witness location.
-    pub fn spend(&mut self, note: &Note, witness: &dyn WitnessTrait) -> Result<(), IronfishError> {
-        let proof = SpendParams::new(self.spender_key.clone(), note, witness)?;
-
-        self.binding_signature_key += proof.value_commitment.randomness;
-        self.binding_verification_key += proof.value_commitment();
-
-        self.spends.push(proof);
+    pub fn add_spend(&mut self, note: Note, witness: &dyn WitnessTrait) {
         self.transaction_fee += note.value() as i64;
```

Comment:
every spends value is added to the transaction_fee variable, weird.

Reply:
Yeah agreed, it should be called `value_balance` or something. We'll be refactoring that soon when adding the asset identifier as part of value.